src/main/java/org/generationcp/middleware/dao/ims/TransactionDAO.java

```diff
@@ -648,16 +654,30 @@ private String buildSearchTransactionsQuery(final TransactionsSearchDto transact
 
 			if (transactionsSearchDto.getTransactionType() != null) {
 				query.append("and (CASE"
-					+ "        WHEN trnstat = " + TransactionStatus.COMMITTED.getIntValue() + " AND trnqty >= 0 THEN '"
+					+ "        WHEN trntype = " + TransactionType.DEPOSIT.getId() + " THEN '"
 					+ TransactionType.DEPOSIT.getValue() + "'"
-					+ "        WHEN trnstat = " + TransactionStatus.ANTICIPATED.getIntValue() + " AND trnqty < 0 THEN '"
-					+ TransactionType.RESERVATION.getValue() + "'"
-					+ "        WHEN trnstat = " + TransactionStatus.COMMITTED.getIntValue() + " AND trnqty < 0 THEN '"
+					+ "        WHEN trntype = " + TransactionType.WITHDRAWAL.getId() + " THEN '"
 					+ TransactionType.WITHDRAWAL.getValue() + "'"
+					+ "        WHEN trntype = " + TransactionType.DISCARD.getId() + " THEN '"
+					+ TransactionType.DISCARD.getValue() + "'"
+					+ "        WHEN trntype = " + TransactionType.ADJUSTMENT.getId() + " THEN '"
+					+ TransactionType.ADJUSTMENT.getValue() + "'"
 					+ "    END) = '")
 					.append(transactionsSearchDto.getTransactionType()).append("' ");
 			}
 
+			if (transactionsSearchDto.getTransactionStatus() != null) {
```

Comment:
transactionStatus in transactionsSearchDto should become an Integer (or List of Integer if we want to be more flexible) so this fragment of query will be simpler

Reply:
Done